cmd/node/node.go

```diff
@@ -1098,9 +1125,13 @@ func (app *SpacemeshApp) Start(*cobra.Command, []string) {
 
 	// app blocks until it receives a signal to exit
 	// this signal may come from the node or from sig-abort (ctrl-c)
-	<-ctx.Done()
+	select {
+	case <-ctx.Done():
+	case err = <-pprofErr:
+	}
 	events.ReportError(events.NodeError{
 		Msg:   "node is shutting down",
 		Level: zapcore.InfoLevel,
 	})
+	return err
```

Comment:
What does this return in the `<-ctx.Done()` case?

Reply:
nil

Reply:
Yes, I see that now, but in this case `err` still holds the value it got on line 1113 above, right? It would be good to make this safer and more explicit by using a different variable name, or else using `err := ...` syntax (rather than `err = ...`). I say "safer" because someone could add code above this that sets `err` to something else, later.

Reply:
To make this more readable, consider something like:

```
	reportErr := func() {
		events.ReportError(events.NodeError{
			Msg:   "node is shutting down",
			Level: zapcore.InfoLevel,
		})
	}
	select {
	case <-ctx.Done():
		reportErr()
		return nil
	case err := <-pprofErr:
		reportErr()
		return err
	}
```